From f5f367d4a9872453888da79bdff3a50f78a9758a Mon Sep 17 00:00:00 2001
From: Ivan Shapovalov <intelfx@intelfx.name>
Date: Wed, 30 Aug 2017 19:49:07 +0300
Subject: [PATCH] cryptsetup-generator: do not bind to the decrypted device
 unit (#6538)

This breaks things when the decrypted device is not immediately
`SYSTEMD_READY=1` (e. g. when a multi-device btrfs system is placed on
multiple cryptsetup devices).

Fixes #6537.

(cherry picked from commit e9ea4526a3a3b41eced29b8d742498cc36750424)
---
 src/cryptsetup/cryptsetup-generator.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/cryptsetup/cryptsetup-generator.c b/src/cryptsetup/cryptsetup-generator.c
index b58b6db7c9..f737f82b55 100644
--- a/src/cryptsetup/cryptsetup-generator.c
+++ b/src/cryptsetup/cryptsetup-generator.c
@@ -109,7 +109,6 @@ static int create_disk(
               "SourcePath=/etc/crypttab\n"
               "DefaultDependencies=no\n"
               "Conflicts=umount.target\n"
-              "BindsTo=dev-mapper-%i.device\n"
               "IgnoreOnIsolate=true\n"
               "After=cryptsetup-pre.target\n",
               f);
